src/browser/uuid_availability.ts

```diff
@@ -0,0 +1,22 @@
+const electron = require('electron');
+const namedMutex = electron.namedMutex;
+import { meshEnabled } from './connection_manager';
+import { getAppRunningState } from './core_state';
+
+const makeMutexKey = (uuid: string) => `uuid-${uuid}`;
+
+export function isUuidAvailable(uuid: string) {
```

Comment:
just stylistically, i would use a stronger verb in the function name here. `isSomething` appears to be just passive "state check" call, but internally this function invokes `tryLock` which affects state globally across the entire system. a stronger name might be `tryReserveUuid` or something.